src/hid_unix.c

```diff
@@ -66,9 +66,16 @@ fido_hid_unix_wait(int fd, int ms, const fido_sigset_t *sigmask)
 		ts.tv_nsec = (ms % 1000) * 1000000;
 	}
 
-	if ((r = ppoll(&pfd, 1, ms > -1 ? &ts : NULL, sigmask)) < 1) {
-		if (r == -1)
-			fido_log_error(errno, "%s: ppoll", __func__);
+	do {
+		r = ppoll(&pfd, 1, ms > -1 ? &ts : NULL, sigmask);
+	} while (r == -1 && errno == EINTR);
+
+	if (r < 0) {
+		fido_log_error(errno, "%s: ppoll", __func__);
+		return (-1);
+	}
+	if (r == 0) {
+		fido_log_error(ETIME, "%s: ppoll timed out", __func__);
 		return (-1);
 	}
```

Comment:
```suggestion
        if (r < 1) {
		if (r == -1)
			fido_log_error(errno, "%s: ppoll", __func__);
		return (-1);
        }
```
This was my original suggestion, which does not print an error if `r == 0`, same as the orginal code.